safe-core-sdk/protocol-kit/migrating/v2.md

```diff
@@ -0,0 +1,48 @@
+# Protocol Kit: Migrating from v1
+
+This guide aims to be a reference of the major changes between v1 and v2 to help those migrating an existing app.
+
+## MasterCopy to Singleton
+
+To avoid confusion between terms that have been used as synonyms we aligned all our code to use the word `singleton`.
+
+- Rename `isL1SafeMasterCopy` to `isL1SafeSingleton`
+```js
+// old:
+SafeFactory.create({ ethAdapter, isL1SafeMasterCopy: true })
+
+// new:
+SafeFactory.create({ ethAdapter, isL1SafeSingleton: true })
+```
+
+## Ethers v6
+
+From `protocolKit v2`, EthersAdapter will only be compatible with ethers.js v6. If you still need to use v5 we recommend you to keep `protocolKit v1`, but we encourage you to migrate to the latest version when you have the chance.
+
+## Protocol Kit createTransaction() accepts only transaction array
+
+In `protocolKit v1` the `createTranasction()` method accepted either an object or an array as parameter. To avoid confusion we aligned to accept only an array. Here is a migration example:
```

Comment:
```suggestion
In `protocolKit v1`, the `createTransaction()` method accepted either an object or an array as a parameter. To avoid confusion, we changed it to accept only an array. Here is a migration example:
```